src/bootstrap/metrics.rs

```diff
@@ -237,13 +265,41 @@ enum JsonNode {
 
         children: Vec<JsonNode>,
     },
-    Test {
-        name: String,
-        #[serde(flatten)]
-        outcome: TestOutcome,
+    TestSuite(TestSuite),
+}
+
+#[derive(Serialize, Deserialize)]
+struct TestSuite {
+    metadata: TestSuiteMetadata,
+    tests: Vec<Test>,
+}
+
+#[derive(Serialize, Deserialize)]
+#[serde(tag = "kind", rename_all = "snake_case")]
+pub(crate) enum TestSuiteMetadata {
+    Crate {
+        crates: Vec<String>,
```

Comment:
the combination of "crate" and "crates" seems confusing - let's call this Package instead (or maybe CargoPackage if you think that's less confusing?)
```suggestion
    Package {
        crates: Vec<String>,
```